singer_sdk/helpers/_flattening.py

```diff
@@ -376,6 +378,11 @@ def flatten_record(
     Returns:
         A flattened version of the record.
     """
+    if flattened_schema is None and max_level is None:
```

Comment:
I wonder the type annotation is accurate... In which cases would `flattened_schema` be `None`?

The only place where this function is called is https://github.com/meltano/sdk/blob/5734e694994562830a5f7d4894910f2a41ec5aac/singer_sdk/mapper.py#L119-L124

and there both `flattened_schema` and `max_level` are ensured to be non-null AFAICT.

Reply:
> I wonder the type annotation is accurate... In which cases would flattened_schema be None?

Yes, makes sense. Let me remove this check and the None option for `flattened_schema`.

Reply:
Maybe it's better to create a flag to enforce the schema on the output or the `max_level` here will be ignored. Wdyt?